### PR TITLE
Add support for the conversion between kind and OpenYurt cluster

### LINKED ISSUE #230: Add support for the conversion between kind cluster and OpenYurt cluster.
OpenYurt supports the conversion between the OpenYurt cluster and the Kubernetes cluster created by minikube, kubeadm, and ack. It may be necessary to add support for the conversion between kind cluster and OpenYurt cluster.

### PR DESCRIPTION
**Pull request by Peeknut**
### Ⅰ. Describe what this PR does

1.Only one node in the cluster is installed with openyurt, and then any node in the cluster can be converted.

2.Add support for the conversion between kind and OpenYurt cluster.

### Ⅱ. Does this pull request fix one issue?

fixes #230.

### Ⅳ. Describe how to verify it

make test
make build
cp _output/bin/yurtctl /bin/
yurtctl convert
yurtctl revert

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/yurtctl/constants/constants.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ spec:
         - /bin/sh
         - -c
         args:
-        - "nsenter -t 1 -m -u -n -i -- /bin/yurtctl convert edgenode --yurthub-image {{.yurthub_image}} --join-token {{.joinToken}}"
+        - "cp /usr/local/bin/yurtctl /tmp && nsenter -t 1 -m -u -n -i -- /var/tmp/yurtctl convert edgenode --yurthub-image {{.yurthub_image}} --join-token {{.joinToken}} && rm /tmp/yurtctl"
```
Comment (charleszheng44) [inline pkg/yurtctl/constants/constants.go:189]: can we execute the `/usr/local/bin/yurtctl` directly?

File: pkg/yurtctl/constants/constants.go [hunk 2]
```diff
@@ -231,7 +231,7 @@ spec:
         - /bin/sh
         - -c
         args:
-        - "nsenter -t 1 -m -u -n -i -- /bin/yurtctl revert edgenode"
+        - "cp /usr/local/bin/yurtctl /tmp && nsenter -t 1 -m -u -n -i -- /var/tmp/yurtctl revert edgenode && rm /tmp/yurtctl"
```
Comment (charleszheng44) [inline pkg/yurtctl/constants/constants.go:234]: Same as the previous comment.

File: pkg/yurtctl/util/edgenode/util.go [hunk 3]
```diff
@@ -113,10 +113,8 @@ func GetNodeName() (string, error) {
 	}
 
 	//2. find --hostname-override in 10-kubeadm.conf
-	nodeName, err := GetSingleContentFromFile(KubeletSvcPath, KubeletHostname)
-	if err != nil {
-		return "", err
-	} else if nodeName != "" {
+	nodeName, _ := GetSingleContentFromFile(KubeletSvcPath, KubeletHostname)
+	if nodeName != "" {
```
Comment (charleszheng44) [inline pkg/yurtctl/util/edgenode/util.go:117]: Can we log the error instead of ignoring it?

File: pkg/yurtctl/util/edgenode/util.go [hunk 4]
```diff
@@ -128,10 +126,8 @@ func GetNodeName() (string, error) {
 	}
 	for _, ef := range environmentFiles {
 		ef = strings.Split(ef, "-")[1]
-		nodeName, err = GetSingleContentFromFile(ef, KubeletHostname)
-		if err != nil {
-			return "", err
-		} else if nodeName != "" {
+		nodeName, _ = GetSingleContentFromFile(ef, KubeletHostname)
+		if nodeName != "" {
```
Comment (charleszheng44) [inline pkg/yurtctl/util/edgenode/util.go:130]: Same as the previous comment.

File: pkg/yurtctl/util/kubernetes/util.go [hunk 5]
```diff
@@ -60,7 +60,7 @@ const (
 
 var (
 	// PropagationPolicy defines the propagation policy used when deleting a resource
-	PropagationPolicy = metav1.DeletePropagationForeground
+	PropagationPolicy = metav1.DeletePropagationBackground
```
Comment (charleszheng44) [inline pkg/yurtctl/util/kubernetes/util.go:63]: Why replace `foreground` with `background`? Shouldn't we want to make sure all dependents are garbage collected? Will `backgound` cause orphan dependents?

File: hack/lib/release-images.sh [hunk 6]
```diff
@@ -83,21 +83,46 @@ function build_multi_arch_binaries() {
...
+                      exit 1
+                 esac
+                 cat << EOF > $docker_file_path
+FROM ${base_image}
+ADD ${binary_name} /usr/local/bin/yurtctl
+EOF
+               else
+                 yurt_component_image="${REPO}/${binary_name}:${TAG}-${arch}"  # openyurt/${binary_name}:latest-${arch}
+                 base_image="k8s.gcr.io/debian-iptables-${arch}:v11.0.2"
+                 cat <<EOF > "${docker_file_path}"
```
Comment (charleszheng44) [inline hack/lib/release-images.sh:120]: nit: This change looks like a generic refactoring, which can be submitted in an independent PR.

**Review by charleszheng44 — CHANGES_REQUESTED** [review]

File: pkg/yurtctl/constants/constants.go (hunk 1 above)
Comment (Peeknut) [reply pkg/yurtctl/constants/constants.go:189]: The operations need to modify the files on the Convert/RevertServant Pod host, so we need to mount yurtctl to the host and run it in the host environment. If run yurtctl directly in the container, the content of the file in the pod is modified.

File: pkg/yurtctl/util/kubernetes/util.go (hunk 5 above)
Comment (Peeknut) [reply pkg/yurtctl/util/kubernetes/util.go:63]: `foreground` needs to wait for all dependents to be collected before deleting the owner.

In kind test, it may take too long to collect the dependents. When the next convert is executed, the resources created last time still exist, so the conversion cannot be performed this time.

`background` will delete the owner first, and then the garbage collector will delete its dependencies in the background.

File: hack/lib/release-images.sh (hunk 6 above)
Comment (Peeknut) [reply hack/lib/release-images.sh:120]: This change is related to the changes in the pkg/yurtctl/constants/constants.go, maybe it's better in a pr?

**Review by rambohe-ch — APPROVED** [review]

File: pkg/yurtctl/constants/constants.go (hunk 1 above)
Comment (charleszheng44) [reply pkg/yurtctl/constants/constants.go:189]: That makes sense. Thanks for explaining

**Review by charleszheng44 — APPROVED** [review]
